pennylane/gradients/param_shift_hessian.py

```diff
@@ -0,0 +1,298 @@
+        hessian = qml.math.transpose(hessian, axes=reordered_axes)
+
+        return qml.math.squeeze(hessian)
+
+    return hessian_tapes, processing_fn
+
+
+@hessian_transform
+def param_shift_hessian(tape, f0=None):
+    r"""Transform a QNode to compute the parameter-shift Hessian with respect to its trainable
+    parameters.
+
+    Args:
+        tape (pennylane.QNode or .QuantumTape): quantum tape or QNode to differentiate
+        f0 (tensor_like[float] or None): Output of the evaluated input tape. If provided,
+            and the Hessian tapes include the original input tape, the 'f0' value is used
+            instead of evaluating the input tape, reducing the number of device invocations.
+
+    Returns:
+        tensor_like or tuple[list[QuantumTape], function]:
+
+        - If the input is a QNode, a tensor representing the output of the hybrid Hessian matrix
+          of size ``(QNode output dimensions, QNode input dimensions, QNode input dimensions)``
+          is returned. When the keyword ``hybrid=False`` is specified, the purely quantum Hessian
+          matrix is returned instead, with the dimensions
+          ``(QNode output dimensions, number of gate arguments, number of gate arguments)``.
+          The difference between the two accounts for the mapping of QNode arguments
+          to the actual gate arguments, which can include classical computations.
+
+        - If the input is a tape, a tuple containing a list of generated tapes, in addition
+          to a post-processing function to be applied to the evaluated tapes.
+
+    **Examples**
+
+        Applying the Hessian transform to a QNode directly computes the Hessian matrix:
+
+        >>> dev = qml.device("default.qubit", wires=2)
+        >>> @qml.qnode(dev)
+        ... def circuit(x):
+        ...     qml.RX(x[0], wires=0)
+        ...    qml.RY(x[1], wires=0)
+        ...    return qml.expval(qml.PauliZ(0))
+        >>> x = np.array([0.1, 0.2], requires_grad=True)
+        >>> qml.gradients.param_shift_hessian(circuit)(x)
+        tensor([[-0.97517033,  0.01983384],
+                [ 0.01983384, -0.97517033]], requires_grad=True)
+
+        Applying it to a quantum tape instead produces the parameter-shifted Hessian tapes
+        and a post-processing function to combine execution results into the Hessian matrix:
+
+        >>> tape = circuit.qnode
+        >>> hessian_tapes, postproc_fn = qml.gradients.param_shift_hessian(circuit)
+        >>> hessian_tapes
+        [<JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>,
+         <JacobianTape: wires=[0], params=2>]
+        >>> postproc_fn(qml.execute(hessian_tapes, dev, None))
+        array([[-0.97517033,  0.01983384],
+               [ 0.01983384, -0.97517033]])
+    """
```

Comment:
very nice examples! You could also stick the drawing example inside a `.. UsageDetails::` section :)

Reply:
Done :)